src/frontend/screens/WebView/index.tsx

```diff
@@ -124,6 +125,18 @@ function WebView() {
     window.api.trackScreen('WebView', { url: startUrl, runner })
   }, [startUrl, runner])
 
+  useEffect(() => {
+    if (!urlIsHpUrl(startUrl)) {
+      return 
+    }
+
+    window.api.handleGoToGamePage(async (_, gameId) => {
+      const gameInfo = await getGameInfo(gameId, 'hyperplay')
+      navigate(`/gamepage/hyperplay/${gameId}`, { state: { gameInfo, fromDM: false } })
+    }
+    )
+  }, [])
```

Comment:
```suggestion
    const removeHandleGoToGamePage = window.api.handleGoToGamePage(async (_, gameId) => {
      const gameInfo = await getGameInfo(gameId, 'hyperplay')
      navigate(`/gamepage/hyperplay/${gameId}`, { state: { gameInfo, fromDM: false } })
    }
    )
    
    //useEffect unmount
    return () => {
      removeHandleGoToGamePage()
    }
  }, [])
```

need to unmount or else it will add another listener on every render of the webview

Reply:
Good catch, We do that on the api side so I thought we didn't need it here as well just like we are doing with the handleGameStatus, but makes sense to have it here as well.

Reply:
done